src/middlewared/middlewared/plugins/directoryservices_/activedirectory_join_mixin.py

```diff
@@ -192,23 +250,20 @@ def setspn(spn):
                 self.logger.error("%s: failed to set spn entry: %s", spn,
                                   netads.stdout.decode().strip())
 
-        setspn(f'nfs/{netbiosname.upper()}')
-        setspn(f'nfs/{netbiosname.upper()}.{domainname.lower()}')
+        setspn(f'nfs/{hostname.upper()}')
+        setspn(f'nfs/{hostname.upper()}.{domainname.lower()}')
         self.middleware.call_sync('kerberos.keytab.store_ad_keytab')
 
     @kerberos_ticket
     def _ad_test_join(self, ds_type: DSType, domain: str):
```

Comment:
`ds_type` is not used and could be removed.

If removed, then do the same with `_ipa_test_join` in `ipa_join_mixin.py` and fixup the call in `connection.py`
```suggestion
    def _ad_test_join(self, domain: str):
```